docs/testplan/ecn/WRED-and-ECN-statistics-testplan.md

```diff
@@ -0,0 +1,254 @@
+             "green_drop_probability": "100",
+             "green_max_threshold": "2240000",
+             "green_min_threshold": "1120000",
+             "red_drop_probability": "100",
+             "red_max_threshold": "1344000",
+             "red_min_threshold": "672000",
+             "wred_green_enable": "true",
+             "wred_red_enable": "true",
+             "wred_yellow_enable": "true",
+             "yellow_drop_probability": "100",
+             "yellow_max_threshold": "1792000",
+             "yellow_min_threshold": "896000"
+         }
+      }
+     "QUEUE": {
+         "Ethernet252|0": {
+             "wred_profile": "[WRED_PROFILE|AZURE_LOSSY]"
+         }
+     }
+ ### **Related SAI APIs and Attributes**
+ -------------------------------
+ Following SAI statistics will get tested:
+
+ SAI_PORT_STAT_GREEN_WRED_DROPPED_PACKETS <br>
+ SAI_PORT_STAT_YELLOW_WRED_DROPPED_PACKETS <br>
+ SAI_PORT_STAT_RED_WRED_DROPPED_PACKETS <br>
+ SAI_PORT_STAT_WRED_DROPPED_PACKETS <br>
+ SAI_QUEUE_STAT_WRED_ECN_MARKED_PACKETS <br>
+ SAI_QUEUE_STAT_WRED_ECN_MARKED_BYTES <br>
+
+ <span id="_Toc463421033" class="anchor"></span>
+ ## **Test methodology**
+
+ 1. Apply WRED/ECN Profile to the queue  .
+
+ 2. Query STATE_DB using REDIS CLI to find out if platform supports WRED/ECN  .
+
+ 3. On supported platform:
+
+    Enable/Disable WRED at queue level using config:<br>
+        counterpoll wred-queue enable/disable<br>
+    Enable/Disable WRED at interface level using config:
+        counterpoll wred-port enable/disable<br>
+
+ 4. on egress port  set   attribute SAI_PORT_ATTR_PKT_TX_ENABLE = false . <br>
+    ( It will build up the traffic in egress queue thereby congestion gets simulated and DUT reacts with WRED/ECN if configured )
+
+ 5. Reset counters using the CLI:<br>
+        sonic-clear queue wredcounters
+
+ 6. Send TCP/UDP traffic switched/routed to the egress interface which is in tx disable state .
+
+ 7. on egress port  set   attribute SAI_PORT_ATTR_PKT_TX_ENABLE = true . <br>
+
+ 8. on supported platform:<br>
+      Verify WRED/ECN counters at queue level using the  new CLI:<br>
+        show queue wredcounters Ethernet&lt;&gt;
+
+      Verify WRED statistics at port level using the enhanced CLI:<br>
+        show interfaces counters detailed Ethernet&lt;&gt;
+
+
+
+
+ Test cases
+ ----------
+ WRED/ECN statistics verification on supported platform
+
+ Terminology:<br>
+    WRED=0/1 refers WRED disabled/enabled on queue <br>
+    ECN=0/1 refers ECN disabled/enabled queue <br>
+    CODE POINT refers to ECN CODE POINT in the IP header <br>
+ | <div style="width:250px">Testcase</div> | <div style="width:300px">Test expectation </div>                                               | <div style="width:500px">Test Steps</div> |
+ | --------------------------------------- | -----------------------------------------------------------------------------------------------| ----------------------------------------- |
```

Comment:
It seems that we lack the test for WRED=0 ECN=1?  Can we add some tests for this scenarios?